hybridCarboniteMetadataService/src/test/scala/cromwell/services/metadata/hybridcarbonite/CarbonitingMetadataFreezerActorSpec.scala

```diff
@@ -80,7 +81,141 @@ class CarbonitingMetadataFreezerActorSpec extends TestKitSuite("CarbonitedMetada
+    val actor = TestFSMRef(new TestableCarbonitingMetadataFreezerActor(carboniterConfig, carboniteWorkerActor.ref, serviceRegistryActor.ref, ioActor.ref))
+    val workflowIdToFreeze = WorkflowId.randomId()
+
+    actor ! FreezeMetadata(workflowIdToFreeze)
+
+    serviceRegistryActor.expectMsgPF(10.seconds) {
+      case gma: GetMetadataAction => gma.workflowId should be(workflowIdToFreeze)
+    }
+    eventually {
+      actor.stateName should be(Fetching)
+    }
+
+    // Freezer actor will receive `FailedMetadataJsonResponse` from service registry actor on any failure, including DB read timeout
+    serviceRegistryActor.send(actor, FailedMetadataJsonResponse(null, new RuntimeException("Cannot read classic metadata from DB")))
+
+    eventually {
+      actor.underlyingActor.updateArchiveStatusCall should be((workflowIdToFreeze, MetadataArchiveStatus.ArchiveFailed))
+      actor.stateName should be(UpdatingDatabase)
+    }
+
+    // Finally, when the database responds appropriately, the actor should shut itself down:
+    watch(actor)
+    actor.underlyingActor.updateArchiveStatusPromise.success(1)
+    eventually {
+      actor.stateName should be(Pending)
+    }
+  }
+
+  it should "when status update in DB fails retry it until success" in {
+    val actor = TestFSMRef(new TestableCarbonitingMetadataFreezerActor(carboniterConfig, carboniteWorkerActor.ref, serviceRegistryActor.ref, ioActor.ref))
+    val workflowIdToFreeze = WorkflowId.randomId()
+
+    actor ! FreezeMetadata(workflowIdToFreeze)
+
+    serviceRegistryActor.expectMsgPF(10.seconds) {
+      case gma: GetMetadataAction => gma.workflowId should be(workflowIdToFreeze)
+    }
+    eventually {
+      actor.stateName should be(Fetching)
+    }
+
+    val jsonValue =
+      s"""{
+         |  "id": "$workflowIdToFreeze",
+         |  "status": "Successful"
+         |}""".stripMargin.parseJson
+
+    serviceRegistryActor.send(actor, SuccessfulMetadataJsonResponse(null, jsonValue.asJsObject))
+
+    var ioCommandPromise: Promise[Any] = null
+    ioActor.expectMsgPF(10.seconds) {
+      case command @ IoCommandWithPromise(ioCommand: IoWriteCommand, _) =>
+        ioCommand.file.pathAsString should be(HybridCarboniteConfig.pathForWorkflow(workflowIdToFreeze, "carbonite-test-bucket"))
+        ioCommand.content should be(jsonValue.prettyPrint)
+        ioCommand.compressPayload should be(true)
+        ioCommandPromise = command.promise
+    }
+    eventually {
+      actor.stateName should be(Freezing)
+    }
+
+    // Simulates the IoActor completing the IO command successfully:
+    ioCommandPromise.success(())
+    eventually {
+      actor.underlyingActor.updateArchiveStatusCall should be((workflowIdToFreeze, MetadataArchiveStatus.Archived))
+      actor.stateName should be(UpdatingDatabase)
+    }
+
+    // When the database update fails, the actor should retry update endlessly until success:
```

Comment:
TOL hmm the "endlessly" is kind of surprising, even if it worked out well for this test...